src/core/dfn.js

```diff
@@ -36,6 +71,68 @@ export function run() {
   }
   sub("plugins-done", addContractDefaults);
 }
+
+/**
+ * @param {HTMLElement} dfn
+ * @param {string} linkingText
+ * */
+function computeTypeAndExport(dfn, linkingText) {
+  let shouldExport = false;
+  let type = "";
+
+  switch (true) {
+    // class defined type (e.g., "<dfn class="element">)
+    case knownTypes.some(name => dfn.classList.contains(name)):
+      // First one wins
+      type = [...dfn.classList].find(className =>
+        knownTypes.includes(className)
+      );
+      validateDefinition(linkingText, type, dfn);
+      shouldExport = true;
+      break;
+
+    // Internal slots: attributes+ methods (e.g., [[some words]](with, optional, arguments))
+    case slotRegex.test(linkingText):
+      shouldExport = false;
+      type = processAsInternalSlot(linkingText, dfn);
+      break;
+  }
+
+  // If the Editor explicitly asked for it to be exported, so let's export it.
+  if (dfn.classList.contains("export")) shouldExport = true;
+
+  if (shouldExport && !dfn.hasAttribute("data-noexport")) {
+    dfn.dataset.export = "";
+  }
```

Comment:
Nit: Let's move this assignment to the bottom of function, so we've all getters together, followed by setters.